ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/redesigned-gas-fees.tsx

```diff
@@ -0,0 +1,520 @@
+            toCurrency: currentCurrency,
+            numberOfDecimals: 2,
+          }),
+        ),
+      )
+    : null;
+
+  // Total
+  const getTransactionFeeTotal = useMemo(() => {
+    return addHexes(gasEstimate, (layer1GasFee as string) ?? 0);
+  }, [gasEstimate, layer1GasFee]);
+
+  const nativeCurrencyTotalFees = layer1GasFee
+    ? getEthConversionFromWeiHex({
+        value: getTransactionFeeTotal,
+        fromCurrency: EtherDenomination.GWEI,
+        numberOfDecimals: 4,
+      })
+    : null;
+
+  const currentCurrencyTotalFees = layer1GasFee
+    ? fiatFormatter(
+        Number(
+          getValueFromWeiHex({
+            value: getTransactionFeeTotal,
+            conversionRate,
+            fromCurrency: EtherDenomination.GWEI,
+            toCurrency: currentCurrency,
+            numberOfDecimals: 2,
+          }),
+        ),
+      )
+    : null;
+
+  return (
+    <>
+      {hasLayer1GasFee && (
+        <Layer2GasFeesExpandBtn
+          expandFeeDetails={expandFeeDetails}
+          setExpandFeeDetails={setExpandFeeDetails}
+        />
+      )}
+      {hasLayer1GasFee && expandFeeDetails && (
+        <>
+          <Layer2FeesRow
+            currentCurrencyFees={currentCurrencyFees}
+            nativeCurrencyFees={nativeCurrencyFees}
+          />
+          <Layer1FeesRow
+            currentCurrencyL1Fees={currentCurrencyL1Fees}
+            nativeCurrencyL1Fees={nativeCurrencyL1Fees}
+          />
+          <TotalFeesRow
+            currentCurrencyTotalFees={currentCurrencyTotalFees}
+            nativeCurrencyTotalFees={nativeCurrencyTotalFees}
+          />
+        </>
+      )}
+    </>
+  );
+};
+
+const TotalGasFees = ({
+  nativeCurrencyFees,
+}: {
+  nativeCurrencyFees: string | undefined;
+}) => {
+  return (
+    <ConfirmInfoRow
+      label="Total"
+      variant={ConfirmInfoRowVariant.Default}
+      tooltip="total tooltip"
+    >
+      <Text>{nativeCurrencyFees}</Text>
+    </ConfirmInfoRow>
+  );
+};
+
+const Divider = () => (
+  <Box
+    borderColor={BorderColor.borderMuted}
+    borderWidth={1}
+    width={BlockSize.Full}
+  />
+);
+
+const GasTimings = ({
+  maxFeePerGas,
+  maxPriorityFeePerGas,
+}: {
+  maxFeePerGas: string;
+  maxPriorityFeePerGas: string;
+}) => {
+  return (
+    <ConfirmInfoRow label="Gas speed" variant={ConfirmInfoRowVariant.Default}>
+      <Box display={Display.Flex} alignItems={AlignItems.center}>
+        {/* TODO: Fix bug in the gas timing component after selection is made */}
+        <GasTiming
+          maxFeePerGas={maxFeePerGas}
+          maxPriorityFeePerGas={maxPriorityFeePerGas}
+        />
+      </Box>
+    </ConfirmInfoRow>
+  );
+};
+
+const EditGasFeesRow = ({
+  currentCurrencyFees,
+  nativeCurrencyFees,
+  supportsEIP1559,
+  setShowCustomizeGasPopover,
+}: {
+  currentCurrencyFees: string;
+  nativeCurrencyFees: string | undefined;
+  supportsEIP1559: boolean;
+  setShowCustomizeGasPopover: Dispatch<SetStateAction<boolean>>;
+}) => {
```

Comment:
IMHO props passing like this and component just to write JSX makes code harder to understand as we need to navigate between child and parent component to get understanding to these props.